File: lib/domain/usecase/save_reading_progress_use_case.dart
import '../../core/model/reading_progress.dart';
import '../repository/book_repository.dart';

class SaveReadingProgressUseCase {
  final BookRepository _bookRepository;

  SaveReadingProgressUseCase(this._bookRepository);

  Future<void> call(ReadingProgress progress) =>
      _bookRepository.saveReadingProgress(progress);
}

File: domain/src/main/kotlin/com/readler/domain/usecase/SaveReadingProgressUseCase.kt
package com.readler.domain.usecase

import com.readler.core.model.ReadingProgress
import com.readler.domain.repository.BookRepository

class SaveReadingProgressUseCase(
    private val bookRepository: BookRepository
) {
    suspend operator fun invoke(progress: ReadingProgress) {
        bookRepository.saveReadingProgress(progress)
    }
}
